apps/server/src/http/routes/chat.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright 2025 BrowserOS
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { Hono } from 'hono'
+import { stream } from 'hono/streaming'
+import { AIProvider } from '../../agent/agent/gemini-vercel-sdk-adapter/types.js'
+import {
+  formatUIMessageStreamDone,
+  formatUIMessageStreamEvent,
+} from '../../agent/agent/gemini-vercel-sdk-adapter/ui-message-stream.js'
+import { AgentExecutionError } from '../../agent/errors.js'
+import type { ChatRequest } from '../../agent/http/types.js'
+import { ChatRequestSchema } from '../../agent/http/types.js'
+import type { RateLimiter } from '../../agent/rate-limiter/index.js'
+import { SessionManager } from '../../agent/session/SessionManager.js'
+import type { Logger } from '../../common/index.js'
+import { Sentry } from '../../common/sentry/instrument.js'
+import { validateRequest } from '../utils/validation.js'
+
+const DEFAULT_MCP_SERVER_URL = 'http://127.0.0.1:9150/mcp'
+const DEFAULT_TEMP_DIR = '/tmp'
+
+interface ChatRouteDeps {
+  logger: Logger
+  tempDir?: string
+  browserosId?: string
+  rateLimiter?: RateLimiter
+}
+
+export function createChatRoutes(deps: ChatRouteDeps) {
+  const { logger, tempDir, browserosId, rateLimiter } = deps
+
+  // Use consolidated server's MCP endpoint
+  const mcpServerUrl = DEFAULT_MCP_SERVER_URL
```

Comment:
because it's now running on the same server, we should just use default MCP port. Why do we even need this?

We should avoid global constants as much as possible.